src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1694,6 +1694,45 @@ MethodTableBuilder::BuildMethodTableThrowing(
         &pByValueClassCache, bmtMFDescs, bmtFP,
         &totalDeclaredFieldSize);
 
+    if (IsValueClass())
+    {
+        const void* pVal;                  // The custom value.
+        ULONG       cbVal;                 // Size of the custom value.
+        HRESULT hr = GetCustomAttribute(bmtInternal->pType->GetTypeDefToken(),
+            WellKnownAttribute::InlineArrayAttribute,
+            &pVal, &cbVal);
+
+        if (hr != S_FALSE)
+        {
+            if (bmtEnumFields->dwNumInstanceFields != 1)
+            {
+                // TODO: diagnostics, must have one field.
+                BuildMethodTableThrowException(IDS_CLASSLOAD_GENERAL);
```

Comment:
should there be a non-generic error here?

Reply:
yes, ideally the same as in the native loader